extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -111,10 +112,26 @@ export class AmpViewerIntegration {
         }
       );
     }
-    /** @type {?HighlightInfoDef} */
-    const highlightInfo = getHighlightParam(ampdoc);
-    if (highlightInfo) {
-      this.highlightHandler_ = new HighlightHandler(ampdoc, highlightInfo);
+
+    if (
+      'fragmentDirective' in document &&
+      isExperimentOn(this.win, 'enable-text-fragments')
+    ) {
+      this.win.addEventListener('message', (e) => {
+        if (
+          e.origin !== 'https://www.google.com' ||
+          e.data?.directive?.length === 0
+        ) {
+          return;
+        }
+        this.updateUrlWithTextFragment_(e.data.directive);
+      });
+    } else {
+      /** @type {?HighlightInfoDef} */
+      const highlightInfo = getHighlightParam(ampdoc);
+      if (highlightInfo) {
+        this.highlightHandler_ = new HighlightHandler(ampdoc, highlightInfo);
```

Comment:
Why aren't we just taking advantage of the already coded `highlight` param, and just using the native fragment handler?

Reply:
Good point. Perhaps we can use this approach & avoid code changes on GWS.

Reply:
Updated to use this approach.